### PR TITLE
Database and persistence. 

### PR DESCRIPTION
**Pull request by hhualiang**
Database and persistence implemented.
Local movie database added.
Local genre database added.
Now the app works without connection.
Tests added for database. 

https://github.com/hhualiang/themoviedb/assets/113935627/74e6972f-33e6-4115-a701-32b8f0358a3c

https://github.com/hhualiang/themoviedb/assets/113935627/b1af2591-481c-42af-8dfd-364130057353

### CODE REVIEW

**Review by MarcosCGlobant — APPROVED** [review]

**Review by matiasilveiro-gl — APPROVED** [review]

File: lib/src/data/repository/movie_repository.dart [hunk 1]
```diff
@@ -35,7 +35,7 @@ class MovieRepository extends IMovieRepository {
         await moviesApiProvider.fetchMovieList(category);
     if (result.state == BaseState.success) {
       return DataSuccess<MovieModel>(
-        MovieModel.fromJson(result.data),
+        MovieModel.fromJson(result.data, category),
```
Comment (FranciscoGaraventa) [inline lib/src/data/repository/movie_repository.dart]: trailing comma

File: lib/src/core/util/dependency_initializer.dart [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+  late MovieDatabase _movieDatabase;
+  late MovieController _movieController;
+  late MovieGenresController _genresController;
+  late MovieRepository _movieRepository;
+  late DatabaseRepository _databaseBaseRepository;
+
+  Future<void> initialize() async {
+    _movieRepository = MovieRepository.instance();
+    _movieDatabase =
+        await $FloorMovieDatabase.databaseBuilder('movie_database.db').build();
```
Comment (FranciscoGaraventa) [inline lib/src/core/util/dependency_initializer.dart]: add value to const

File: lib/src/domain/entity/movie.dart [hunk 3]
```diff
@@ -31,24 +36,34 @@ class Movie {
     required this.hasVideo,
     required this.movieRating,
     required this.voteCount,
+    required this.category,
   });
 
-  factory Movie.fromJson(Map<String, dynamic> json) {
+  factory Movie.fromJson(Map<String, dynamic> json, String category) {
```
Comment (FranciscoGaraventa) [inline lib/src/domain/entity/movie.dart]: trailing comma

File: lib/src/presentation/widget/home_screen/home_screen_movies_stream_builder.dart [hunk 4]
```diff
@@ -26,10 +30,12 @@ class MovieStreamBuilder extends StatelessWidget {
...
           case BaseState.empty:
-            return const SizedBox.shrink();
+            return const Center(
+              child: Text(_emptyMessage),
+            );
           case BaseState.failure:
-            return Center(
-              child: Text(snapshot.data!.error!),
+            return const Center(
+              child: Text(_failedMessage, textAlign: TextAlign.center,),
```
Comment (FranciscoGaraventa) [inline lib/src/presentation/widget/home_screen/home_screen_movies_stream_builder.dart]: format

**Review by FranciscoGaraventa — APPROVED** [review]
Approve with minor comments

**Pull request merged**